ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -469,11 +475,22 @@ private void onResponse(List<ResponseInfo> responseInfoList) {
      */
     @Override
     public void run() {
+      long startTime;
+      long lastTime;
       try {
         while (isOpen.get()) {
+          startTime = time.milliseconds();
+          lastTime = startTime;
           List<RequestInfo> requestInfoList = pollForRequests();
+          routerMetrics.operationControllerPollforRequestTimeMs.update(time.milliseconds() - lastTime);
+          lastTime = time.milliseconds();
           List<ResponseInfo> responseInfoList = networkClient.sendAndPoll(requestInfoList);
+          routerMetrics.numOfRequestInfosPerPoll.update(requestInfoList.size());
+          routerMetrics.operationControllerSendAndPollTimeMs.update(time.milliseconds() - lastTime);
+          lastTime = time.milliseconds();
           onResponse(responseInfoList);
+          routerMetrics.operationControllerProcessResponseTimeMs.update(time.milliseconds() - lastTime);
```

Comment:
> Perhaps we could add a metric for how long handling response takes for each operation manager, within their individual handleResponse() methods.

Why this metric is important? I did not see any time-consuming part within the `handleResponse()` method of each manager. Also each time only a single response gets processed. Compared with i/o and polling time, this time should be very small.